samples/net/sockets/tls_configurations/src/main.c

```diff
@@ -0,0 +1,231 @@
+
+#define INVALID_SOCKET (-1)
+
+/* Keep the new line because openssl uses that to start processing the incoming data */
+#define TEST_STRING		"hello world\n"
+
+static int socket_fd = INVALID_SOCKET;
+static struct pollfd fds[1 + 1];
+static int nfds = 2;
+
+static uint8_t test_buf[128];
+
+static int wait_for_event(void)
+{
+	int ret;
+
+	/* Wait for event on any socket used. Once event occurs,
+	 * we'll check them all.
+	 */
+	ret = poll(fds, nfds, -1);
+	if (ret < 0) {
+		LOG_ERR("Error in poll (%d)", errno);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int create_socket(void)
+{
+	int ret = 0;
+	struct sockaddr_in addr;
+
+	addr.sin_family = AF_INET;
+	addr.sin_port = htons(PEER_PORT);
+	inet_pton(AF_INET, "127.0.0.1", &addr.sin_addr);
+
+	socket_fd = socket(addr.sin_family, SOCK_STREAM, IPPROTO_TLS_1_2);
+	if (socket_fd < 0) {
+		LOG_ERR("Failed to create TLS socket (%d)", errno);
+		return -errno;
+	}
+
+	sec_tag_t sec_tag_list[] = {
+#if defined(SOME_CERTIFICATE_SUPPORTED)
+		CA_CERTIFICATE_TAG,
+#endif
+#if defined(CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+		PSK_TAG,
+#endif
+	};
+
+	ret = setsockopt(socket_fd, SOL_TLS, TLS_SEC_TAG_LIST,
+			sec_tag_list, sizeof(sec_tag_list));
+	if (ret < 0) {
+		LOG_ERR("Failed to set TLS_SEC_TAG_LIST option (%d)", errno);
+		return -errno;
+	}
+
+	/* HOSTNAME is only required for key exchanges that use a certificate. */
+#if defined(SOME_CERTIFICATE_SUPPORTED)
+	ret = setsockopt(socket_fd, SOL_TLS, TLS_HOSTNAME,
+			 "localhost", sizeof("localhost"));
+	if (ret < 0) {
+		LOG_ERR("Failed to set TLS_HOSTNAME option (%d)", errno);
+		return -errno;
+	}
+#endif
+
+	ret = connect(socket_fd, (struct sockaddr *) &addr, sizeof(addr));
+	if (ret < 0) {
+		LOG_ERR("Cannot connect to TCP remote (%d)", errno);
+		return -errno;
+	}
+
+	/* Prepare file descriptor for polling */
+	fds[0].fd = eventfd(0, 0);
+	fds[0].events = POLLIN;
+
+	fds[1].fd = socket_fd;
+	fds[1].events = POLLIN;
+
+	return ret;
+}
+
+void close_socket(void)
+{
+	if (socket_fd != INVALID_SOCKET) {
+		close(socket_fd);
+	}
+}
+
+static int setup_credentials(void)
+{
+	LOG_INF(APP_BANNER);
+	int err;
+
+#if defined(SOME_CERTIFICATE_SUPPORTED)
+	err = tls_credential_add(CA_CERTIFICATE_TAG,
+				TLS_CREDENTIAL_CA_CERTIFICATE,
+				certificate,
+				sizeof(certificate));
+	if (err < 0) {
+		LOG_ERR("Failed to register certificate: %d", err);
+		return err;
+	}
+#endif
+
+#if defined(CONFIG_MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+	err = tls_credential_add(PSK_TAG,
+				TLS_CREDENTIAL_PSK,
+				psk,
+				sizeof(psk));
+	if (err < 0) {
+		LOG_ERR("Failed to register PSK: %d", err);
+	}
+	err = tls_credential_add(PSK_TAG,
+				TLS_CREDENTIAL_PSK_ID,
+				psk_id,
+				sizeof(psk_id) - 1);
+	if (err < 0) {
+		LOG_ERR("Failed to register PSK ID: %d", err);
+	}
+#endif
+
+	return 0;
+}
+
+int main(void)
+{
+	int ret;
+	size_t data_len;
+
+	setup_credentials();
+
+	k_thread_priority_set(k_current_get(), 0);
```

Comment:
Maybe add a comment to explain why this?

Reply:
Sorry, that's another leftover. I removed it.